pkg/cache/cache.go

```diff
@@ -30,26 +32,67 @@ type SharedInformerCache interface {
 
 	// ListApplicationRegistrations lists the application registration CRs from the cache
 	ListApplicationRegistrations() (*storkv1alpha1.ApplicationRegistrationList, error)
+
+	// ListPods lists the all the Pods from the cache
+	ListTransformedPods() (*corev1.PodList, error)
 }
 
 type cache struct {
-	client client.Client
+	controllerCache controllercache.Cache
 }
 
 var (
 	cacheLock           sync.Mutex
 	sharedInformerCache *cache
 )
 
-func CreateSharedInformerCache(mgr manager.Manager) error {
+func CreateSharedInformerCache() error {
 	cacheLock.Lock()
 	defer cacheLock.Unlock()
 	if sharedInformerCache != nil {
 		return fmt.Errorf("shared informer cache already initialized")
 	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	// Define a transform map to modify the informer's objects before they're added to the cache.
+	transformMap := controllercache.TransformByObject{
+		&corev1.Pod{}: func(obj interface{}) (interface{}, error) {
+			podResource, ok := obj.(*corev1.Pod)
+			if !ok {
+				return nil, fmt.Errorf("unexpected object type: %T", obj)
+			}
+			currPod := corev1.Pod{}
+			currPod.Name = podResource.Name
+			currPod.Namespace = podResource.Namespace
+			currPod.Namespace = podResource.Namespace
+
+			currPod.Spec.Volumes = podResource.Spec.Volumes
+			currPod.Spec.Containers = podResource.Spec.Containers
```

Comment:
Do we need Containers?

Reply:
We do not need it in health-monitor which is the only place where we are using this Cache for listing Pods. However, we are using  Containers in webhook and might later want to update that interface to use the cache instead. So I will keep this for now.